src/Interpreters/ITokenExtractor.cpp

```diff
@@ -270,4 +300,46 @@ void SplitTokenExtractor::substringToGinFilter(const char * data, size_t length,
             gin_filter.addTerm(data + token_start, token_len);
 }
 
+std::vector<String> NoOpTokenExtractor::getTokens(const char * data, size_t length) const
+{
+    return {{data, length}};
+}
+
+bool NoOpTokenExtractor::nextInString(const char * /*data*/, size_t length, size_t * __restrict pos, size_t * __restrict token_start, size_t * __restrict token_length) const
+{
+    if (*pos == 0)
+    {
+        *token_start = 0;
+        *token_length = length;
+        return true;
+    }
+    return false;
+}
+
+bool NoOpTokenExtractor::nextInStringLike(const char * /*data*/, size_t /*length*/, size_t * __restrict /*token_start*/, String & /*token_length*/) const
+{
+    return false;
+}
+
+#if USE_CPPJIEBA
+ChineseTokenExtractor::ChineseTokenExtractor(ChineseTokenizationGranularity granularity_)
+    : granularity(granularity_)
+{}
+
+std::vector<String> ChineseTokenExtractor::getTokens(const char * data, size_t /*length*/) const
+{
+    return ChineseTokenizer::instance().tokenize(data, granularity);
+}
+
+bool ChineseTokenExtractor::nextInString(const char * /*data*/, size_t /*length*/, size_t * __restrict /*pos*/, size_t * __restrict /*token_start*/, size_t * __restrict /*token_length*/) const
+{
+    throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Method nextInStringLike is not supported by ChineseTokenExtractor");
```

Comment:
There was some implementation here which didn't look finished. Now throws an exception to be on the safe side.